### PR TITLE
feat(docker): add high severity policies

### PR DESCRIPTION
**Pull request by AndreyLevchenko**

### CODE REVIEW

File: docker/policies/yum_clean_all_missing.rego [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+deny[res] {
+	args := get_yum[_]
+	res := sprintf("'yum clean all' is missed: %s", [args])
+}
+
+contains_clean_after_yum(cmd) {
+	yum_install_command := regex.find_n("yum (-[a-zA-Z]+ *)*install", cmd, -1)
+
+	install := indexof(cmd, yum_install_command[0])
```
Comment (knqyf263) [inline docker/policies/yum_clean_all_missing.rego:38]: I think we should take the last element. It cannot detect as below though it is not practical.

```
RUN yum -y install bash && yum clean all && yum -y install zsh
```

File: docker/policies/run_command_cd_instead_of_workdir_test.rego [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		{
+			"Cmd": "env",
+			"Value": [
+				"AUTHOR",
+				"Docker",
+			],
+		},
+		{
+			"Cmd": "run",
+			"Value": ["cd /usr/share/nginx/html"],
```
[the remark this answers is not in the document]
Comment (AndreyLevchenko) [reply docker/policies/run_command_cd_instead_of_workdir_test.rego:18]: done

File: docker/policies/run_using_wget_and_curl.rego [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+get_tool_usage(cmd, cmd_name) = wget {
+	count(cmd.Value) == 1
+
+	commandsList = split(cmd.Value[0], "&&")
+
+	reg_exp = sprintf("^( )*%s", [cmd_name])
+
+	wget := [x | instruction := commandsList[i]; not contains(instruction, "install "); regex.match(reg_exp, instruction) == true; x := cmd.Value[0]]
```
[the remark this answers is not in the document]
Comment (AndreyLevchenko) [reply docker/policies/run_using_wget_and_curl.rego:38]: done

File: docker/policies/yum_clean_all_missing_test.rego [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+			},
+		],
+		"alpine:3.4": [
+			{
+				"Cmd": "from",
+				"Value": ["alpine:3.4"],
+			},
+			{
+				"Cmd": "run",
+				"Value": ["yum -y install yum clean all"],
```
[the remark this answers is not in the document]
Comment (AndreyLevchenko) [reply docker/policies/yum_clean_all_missing_test.rego:62]: done

File: docker/policies/yum_clean_all_missing.rego (hunk 1 above)
Comment (AndreyLevchenko) [reply docker/policies/yum_clean_all_missing.rego]: done

File: docker/policies/run_command_cd_instead_of_workdir.rego [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	"combine": false,
+	"selector": [{"type": "dockerfile"}],
+}
+
+get_cd[args] {
+	some i, j, k
+	run := docker.run[i]
+	parts = regex.split(`\s+&&\s+`, run.Value[j])
+	startswith(parts[k], "cd ")
+	args := concat(" ", run.Value)
```
Comment (knqyf263) [inline docker/policies/run_command_cd_instead_of_workdir.rego:26]:
```suggestion
	run := docker.run[_]
	parts = regex.split(`\s+&&\s+`, run.Value[_])
	startswith(parts[_], "cd ")
	args := concat(" ", run.Value)
```

File: docker/policies/run_command_cd_instead_of_workdir.rego [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+__rego_input__ := {
+	"combine": false,
+	"selector": [{"type": "dockerfile"}],
+}
+
+get_cd[args] {
+	some i, j, k
+	run := docker.run[i]
+	parts = regex.split(`\s+&&\s+`, run.Value[j])
```
Comment (knqyf263) [inline docker/policies/run_command_cd_instead_of_workdir.rego:24]: For `apt-get install -y vim&&cd /tmp`

```suggestion
	parts = regex.split(`\s*&&\s*`, run.Value[j])
```

File: docker/policies/run_using_wget_and_curl.rego [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+	]
+}
+
+get_tool_usage(cmd, cmd_name) = wget {
+	count(cmd.Value) > 1
+
+	cmd.Value[0] == cmd_name
+
+	wget := [concat(" ", cmd.Value)]
+}
```
Comment (knqyf263) [inline docker/policies/run_using_wget_and_curl.rego:55]: Why is this necessary?

File: docker/policies/run_using_wget_and_curl.rego [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+#chained commands
+get_tool_usage(cmd, cmd_name) = wget {
+	count(cmd.Value) == 1
+
+	commands_list = split(cmd.Value[0], "&&")
+
+	reg_exp = sprintf("^( )*%s", [cmd_name])
+
+	wget := [x |
+		instruction := commands_list[i]
```
Comment (knqyf263) [inline docker/policies/run_using_wget_and_curl.rego:40]:
```suggestion
		instruction := commands_list[_]
```

File: docker/policies/run_using_wget_and_curl.rego [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+#chained commands
+get_tool_usage(cmd, cmd_name) = wget {
+	count(cmd.Value) == 1
+
+	commands_list = split(cmd.Value[0], "&&")
+
+	reg_exp = sprintf("^( )*%s", [cmd_name])
+
+	wget := [x |
```
Comment (knqyf263) [inline docker/policies/run_using_wget_and_curl.rego:39]: It might not be wget

File: docker/policies/run_command_cd_instead_of_workdir.rego (hunk 5 above)
Comment (AndreyLevchenko) [reply docker/policies/run_command_cd_instead_of_workdir.rego:26]: done

File: docker/policies/run_command_cd_instead_of_workdir.rego (hunk 6 above)
Comment (AndreyLevchenko) [reply docker/policies/run_command_cd_instead_of_workdir.rego:24]: done

File: docker/policies/run_using_wget_and_curl.rego (hunk 7 above)
Comment (AndreyLevchenko) [reply docker/policies/run_using_wget_and_curl.rego:55]: done

File: docker/policies/run_using_wget_and_curl.rego (hunk 8 above)
Comment (AndreyLevchenko) [reply docker/policies/run_using_wget_and_curl.rego:40]: done

File: docker/policies/run_using_wget_and_curl.rego (hunk 9 above)
Comment (AndreyLevchenko) [reply docker/policies/run_using_wget_and_curl.rego:39]: done

**Review by knqyf263 — APPROVED** [review]
